etna/models/base.py

```diff
@@ -146,3 +148,82 @@ def _build_models(self):
         self._models = {}
         for segment in self._segments:
             self._models[segment] = deepcopy(self._base_model)
+
+
+class FitAbstractModel(ABC):
+    """Interface for model with fit method."""
+
+    @abstractmethod
+    def fit(self, ts: TSDataset) -> "FitAbstractModel":
+        """Fit model.
+
+        Parameters
+        ----------
+        ts:
+            Dataset with features
+
+        Returns
+        -------
+        result:
+            Model after fit
+        """
+        pass
+
+    @abstractmethod
+    def get_model(self) -> Dict[str, Any]:
+        """Get models for segments.
+
+        Returns
+        -------
+        result:
+            Dictionary with internal model for each segment, e.g. CatBoostRegressor or Ridge:
+            * If model is per-segment, then keys are segments.
+            * If model is multi-segment, then key is "all_segments", because there is only one model.
+        """
+        pass
+
+
+class ForecastAbstractModel(ABC):
+    """Interface for model with forecast method."""
+
+    @abstractmethod
+    def forecast(self, ts: TSDataset) -> TSDataset:
+        """Make predictions.
+
+        Parameters
+        ----------
+        ts:
+            Dataset with features
+
+        Returns
+        -------
+        result:
+            Dataset with predictions
+        """
+        pass
+
+
+class PredictIntervalAbstractModel(ABC):
+    """Interface for model with forecast method that creates prediction interval."""
```

Comment:
This is the interface for the models with built-in prediction intervals, it is better to specify it in the description